core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java

```diff
@@ -490,10 +496,20 @@ private void collectDynamicFilters(StageId stageId, Optional<Set<DynamicFilterId
             return ImmutableSet.of();
         }
 
-        PlanNode planNode = plan.getRoot();
-        Set<DynamicFilterId> innerStageDynamicFilters = intersection(getProducedDynamicFilters(planNode), getConsumedDynamicFilters(planNode));
-        Set<DynamicFilterId> replicatedDynamicFilters = getReplicatedDynamicFilters(planNode);
-        return ImmutableSet.copyOf(intersection(innerStageDynamicFilters, replicatedDynamicFilters));
+        Set<DynamicFilterId> innerStageDynamicFilters = intersection(
+                getSourceDistributionStageProducedDynamicFilters(plan),
+                getConsumedDynamicFilters(plan.getRoot()));
+        return ImmutableSet.copyOf(innerStageDynamicFilters);
+    }
+
+    private static Set<DynamicFilterId> getSourceDistributionStageProducedDynamicFilters(PlanFragment plan)
```

Comment:
I assume this change is just a refactor. You just added an assertion that `producedDynamicFilters` are equal to `replicatedDynamicFilters`. And given `producedDynamicFilters` and `replicatedDynamicFilters` sets are equal, `producedDynamicFilters` x `consumedDymanicFilters` x `replicatedDynamicFilters` = `producedDynamicFilters` x `consumedDymanicFilters`.

Is my understanding correct?